### PR TITLE
[DNM] client: add more debug logs for setxattr

### PR DESCRIPTION
**Pull request by lxbsz**
Just for testing to debug issues#47294.

For: https://tracker.ceph.com/issues/47294
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
@lxbsz please rebase.

**Comment by lxbsz** [comment]
> @lxbsz please rebase.

Done. Thanks.

**Comment by batrick** [comment]
jenkins test docs

**Comment by batrick** [comment]
Fix for this problem has been merged: https://tracker.ceph.com/issues/47294

Thanks @lxbsz

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/47294 is back, reopening this for testing

File: src/client/Client.cc [hunk 1]
```diff
@@ -12124,6 +12147,7 @@ void Client::_setxattr_maybe_wait_for_osdmap(const char *name, const void *value
   // For setting pool of layout, MetaRequest need osdmap epoch.
   // There is a race which create a new data pool but client and mds both don't have.
   // Make client got the latest osdmap which make mds quickly judge whether get newer osdmap.
+  ldout(cct, 1) << __func__ << ":" << __LINE__ << " name " << name << " value " << value << dendl;
```
Comment (batrick) [inline src/client/Client.cc:12150]:
```suggestion
  ldout(cct, 15) << __func__ << ":  name=" << name << dendl;
```

File: src/client/Client.cc [hunk 2]
```diff
@@ -12134,7 +12158,9 @@ void Client::_setxattr_maybe_wait_for_osdmap(const char *name, const void *value
 
     if (r == -ENOENT) {
       bs::error_code ec;
+      ldout(cct, 1) << __func__ << ":" << __LINE__ << " name " << name << " value " << value << dendl;
```
Comment (batrick) [inline src/client/Client.cc:12161]:
```suggestion
      ldout(cct, 20) << __func__ << ": waiting for latest osdmap" << dendl;
```

File: src/client/Client.cc [hunk 3]
```diff
@@ -12134,7 +12158,9 @@ void Client::_setxattr_maybe_wait_for_osdmap(const char *name, const void *value
...
       objecter->wait_for_latest_osdmap(ca::use_blocked[ec]);
+      ldout(cct, 1) << __func__ << ":" << __LINE__ << " name " << name << " value " << value << dendl;
```
Comment (batrick) [inline src/client/Client.cc:12163]:
```suggestion
      ldout(cct, 20) << __func__ << ": finished waiting for latest osdmap: " << ec << dendl;
```

**Review by batrick — CHANGES_REQUESTED** [review]
and drop other debug lines. Then we can merge this.

**Comment by batrick** [comment]
superseded by #38836

**Comment by lxbsz** [comment]
Sorry, missed this.

**Pull request closed without merging**